queue.h

```diff
@@ -112,25 +101,17 @@ class ArrayQueue: public BasicQueue<T1>
 private:
 	T1 * _array;
 	size_t head, tail, size, _capacity;
-
-    /*
-     * Fully implement class:
-     * - declarations here
-     * - definitions in queue.cpp
-     * Only finite size implementation is required
-     */
 };
 
 
 
 template<typename T1>
 inline void ArrayQueue<T1>::enqueue(T1 & x)
 {
-	_array[size] = x;
-	tail = size;
+	_array[tail] = x;
 	size = (size + 1) % _capacity;
-	if (size == head) throw logic_error("Queue overflow");
-
+	tail = size;

```

Comment:
Seems that tail shouldn't be assigned to size, it should be incremented separately